### PR TITLE
Build webapi package and deploy Azure infrastructure in parallel

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
Faster deployments are always nicer.

### Description
Build webapi package and deploy Azure infrastructure in parallel instead of having infrastructure deployment wait for webapi package

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/copilot-chat/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: .github/workflows/copilot-deploy-binaries.yml [hunk 1]
```diff
@@ -52,7 +40,6 @@ jobs:
       AZURE_SUBSCRIPTION_ID: ${{secrets.AZURE_SUBSCRIPTION_ID}}
 
   deploy-frontend:
-    needs: [deploy-infra]
     uses: ./.github/workflows/copilot-deploy-frontend.yml
     with:
       DEPLOYMENT_NAME: ${{needs.deploy-infra.outputs.deployment-id}}
```
Comment (alliscode) [inline .github/workflows/copilot-deploy-binaries.yml:45]: This won't work without having deploy-infra in the needs collection. `{{ needs.deploy-infra...`

File: .github/workflows/copilot-deploy-pipeline.yml [hunk 2]
```diff
@@ -17,9 +17,20 @@ jobs:
   build:
     uses: ./.github/workflows/copilot-build-backend.yml
 
-  int:
-    needs: build
```
Comment (alliscode) [inline .github/workflows/copilot-deploy-pipeline.yml:21]: I'm not sure it's a good idea to deploy the infra before we verify the code builds.

**Review by alliscode — CHANGES_REQUESTED** [review]

File: .github/workflows/copilot-deploy-binaries.yml (hunk 1 above)
Comment (alliscode) [reply .github/workflows/copilot-deploy-binaries.yml:45]: Currently the front-end and back-end deployments use the deployment Id to get some information about the resources so they need a reference to it. In theory you could pass this in as a parameter. However, I'm not sure we want to do this before the build has succeeded.

Comment (glahaye) [reply .github/workflows/copilot-deploy-binaries.yml:45]: Argh. Of course!  Whst did I tell you about GH's pipeline syntax...

Abandonning this PR

**Comment by glahaye** [comment]
Abandoning this dubious change

**Pull request closed without merging**